auto_ml/utils.py

```diff
@@ -41,29 +41,63 @@ def __init__(self, column_descriptions=None):
         self.column_descriptions = column_descriptions
         self.vals_to_del = set([None, float('nan'), float('Inf')])
         self.vals_to_ignore = set(['regressor', 'classifier', 'output', 'ignore'])
-
+        self.tfidfvec=TfidfVectorizer()
 
     def fit(self, X, y=None):
-        return self
 
 
+        inputflag=False
+
+        #Condition check if there is text or nlp field only then do tfidf
+        #follow loop will be excuted only one time , must see if there is any other logic.
+        #currently this is needed becuase this is the only way to get to know if there is any senetence as inputs in columns
+        #TODO alternatively any option from config file would be helpful which will remove this following loop
+        for row in X:
+            for key, val in row.items():
+                column_desciption=self.column_descriptions.get(key)
+                if column_desciption=="text" or column_desciption=="nlp":
+                    inputflag=True
+                    break
+        # must look at an alternate way of doing this
+        if inputflag:
+            input = [] #check if this leads to memory leak??
+            for row in X:
+                for key, val in row.items():
+                    col_desc = self.column_descriptions.get(key)
+                    if col_desc=="text" or col_desc=="nlp":
+                            input.append(val)
+            self.tfidfvec.fit(input)
+            return self
+        else:
+            return self
+
     def transform(self, X, y=None):
         clean_X = []
         deleted_values_sample = []
         deleted_info = {}
 
-        for row in X:
-            clean_row = {}
 
+        ##Todo: transformation taken in a dict of dicts and transforms row by row, must use a dataframe based idea instead
+
+        for row in X:
             for key, val in row.items():
+                clean_row = {}
```

Comment:
Did not see that, thank you for pointing out.
I will correct and resend the PR.